Tasks/Common/codecoverage-tools/utilities.ts

```diff
@@ -0,0 +1,196 @@
+    try {
+        //  return tl.stats(path).isFile();
+        return true;
+    }
+    catch (error) {
+        return false;
+    }
+}
+
+// returns true if given string is null or whitespace.
+export function isNullOrWhitespace(input) {
+    if (typeof input === "undefined" || input == null) {
+        return true;
+    }
+    return input.replace(/\s/g, "").length < 1;
+}
+
+// returns empty string if the given value is undefined or null.
+export function trimToEmptyString(input) {
+    if (typeof input === "undefined" || input == null) {
+        return "";
+    }
+    return input.trim();
+}
+
+// appends given text to file.
+export function appendTextToFileSync(filePath: string, fileContent: string) {
+    if (isFileExists(filePath)) {
+        fs.appendFileSync(filePath, fileContent);
+    } else {
+        throw new Error("File not found: " + filePath);
+    }
+}
+
+// prepends given text to start of file.
+export function prependTextToFileSync(filePath: string, fileContent: string) {
+    if (isFileExists(filePath)) {
+        let data = fs.readFileSync(filePath); // read existing contents into data
+        let fd = fs.openSync(filePath, "w+");
+        let buffer = new Buffer(fileContent);
+        fs.writeSync(fd, buffer, 0, buffer.length, 0); // write new data
+        fs.writeSync(fd, data, 0, data.length, 0); // append old data
+        fs.close(fd);
+    }
+}
+
+// single utility for appending text and prepending text to file.
+export function insertTextToFileSync(filePath: string, prependFileContent?: string, appendFileContent?: string) {
+    if (isFileExists(filePath) && (prependFileContent || appendFileContent)) {
+        let existingData = fs.readFileSync(filePath); // read existing contents into data
+        let fd = fs.openSync(filePath, "w+");
+        let preTextLength = prependFileContent ? prependFileContent.length : 0;
+
+        if (prependFileContent) {
+            let prependBuffer = new Buffer(prependFileContent);
+            fs.writeSync(fd, prependBuffer, 0, prependBuffer.length, 0); // write new data
+        }
+        fs.writeSync(fd, existingData, 0, existingData.length, preTextLength); // append old data
+        if (appendFileContent) {
+            let appendBuffer = new Buffer(appendFileContent);
+            fs.writeSync(fd, appendBuffer, 0, appendBuffer.length, existingData.length + preTextLength);
+        }
+        fs.close(fd);
+    }
+}
+
+// trim the given character if it exists in the end of string.
+export function trimEnd(data: string, trimChar: string) {
+    if (!trimChar || !data) {
+        return data;
+    }
+
+    if (str(data).endsWith(trimChar)) {
+        return data.substring(0, data.length - trimChar.length);
+    } else {
+        return data;
+    }
+}
+
+export function readXmlFileAsJson(filePath: string): Q.Promise<any> {
+    console.log("Reading XML file: " + filePath);
+    return readFile(filePath, "utf-8")
+        .then(convertXmlStringToJson);
```

Comment:
It's easier to read if you use lambads: .then((xml) => return convertXmlToJson(xml))